src/renderer/src/components/song/song-detail/SongControls.tsx

```diff
@@ -0,0 +1,106 @@
+import { Component, createEffect, createSignal, Match, Show, Switch } from "solid-js";
+import { isPlaying, next, previous, togglePlay } from "../../../lib/Music";
+import { isSongUndefined } from "../../../lib/song";
+import Fa from "solid-fa";
+import {
+  faBackwardStep,
+  faForwardStep,
+  faPause,
+  faPlay,
+  faVolumeHigh,
+  faVolumeLow,
+  faVolumeXmark,
+  faAdd,
+  faRandom,
+  faRepeat
+} from "@fortawesome/free-solid-svg-icons";
+import { GLOBAL_ICON_SCALE } from "../../../App";
+import Bar from "../../bar/Bar";
+import IconButton from "@renderer/components/icon-button/IconButton";
+import { localVolume, setLocalVolume, song } from "@renderer/lib/state/song";
+
+type SongControlsProps = {};
+
+const SongControls: Component<SongControlsProps> = () => {
+  const [disable, setDisable] = createSignal(isSongUndefined(song()));
+  const [playHint, setPlayHint] = createSignal("");
+
+  createEffect(() => {
+    const disabled = disable();
+
+    if (disabled) {
+      setPlayHint("");
+      return;
+    }
+
+    setPlayHint(isPlaying() ? "Pause" : "Play");
+  });
+
+  createEffect(() => setDisable(isSongUndefined(song())));
+
+  return (
+    <div class="song-controls">
+      {/* Left part */}
+      <div class="song-controls__left-part">
+        <IconButton>
+          <Switch>
+            <Match when={localVolume() === 0}>
+              <Fa icon={faVolumeXmark} scale={GLOBAL_ICON_SCALE} />
+            </Match>
+            <Match when={localVolume() < 0.5}>
+              <Fa icon={faVolumeLow} scale={GLOBAL_ICON_SCALE} />
+            </Match>
+            <Match when={localVolume() >= 0.5}>
+              <Fa icon={faVolumeHigh} scale={GLOBAL_ICON_SCALE} />
+            </Match>
+          </Switch>
+        </IconButton>
+        <div class="song-controls__volume-bar">
+          <Bar fill={localVolume()} setFill={setLocalVolume} />
```

Comment:
https://github.com/Team-BTMC/osu-radio/pull/2/commits/0fee1ad8773723e3b1487f9bf77dc5125e43a26e clamp this value as in the commit here

`caught DOMException: Failed to set the 'volume' property on 'HTMLMediaElement': The volume provided (-0.00119783) is outside the range [0, 1].`

Reply:
or to be fair don't make the function, as I removed it afterwards, but use the clamp

Reply:
Added the clamping.